src/renderer/state.ts

```diff
@@ -374,36 +373,50 @@ export class AppState {
         parsedValue = newValue;
       }
 
-      if (key in GlobalSetting && key in this) {
+      if (Object.values(GlobalSetting).includes(key)) {
         switch (key) {
-          case 'theme': {
+          case GlobalSetting.theme: {
             this.setTheme(parsedValue as string);
             break;
           }
 
-          case 'acceleratorsToBlock':
-          case 'channelsToShow':
-          case 'electronMirror':
-          case 'environmentVariables':
-          case 'executionFlags':
-          case 'fontFamily':
-          case 'fontSize':
-          case 'gitHubAvatarUrl':
-          case 'gitHubLogin':
-          case 'gitHubName':
-          case 'gitHubToken':
-          case 'hasShownTour':
-          case 'isClearingConsoleOnRun':
-          case 'isEnablingElectronLogging':
-          case 'isKeepingUserDataDirs':
-          case 'isPublishingGistAsRevision':
-          case 'isUsingSystemTheme':
-          case 'knownVersion':
-          case 'localVersion':
-          case 'packageAuthor':
-          case 'packageManager':
-          case 'showObsoleteVersions':
-          case 'showUndownloadedVersions': {
+          case GlobalSetting.hasShownTour: {
+            this['isTourShowing'] = !(parsedValue as boolean);
+            break;
+          }
+
+          // This key is deprecated, so do nothing
+          case GlobalSetting.knownVersion: {
+            break;
+          }
+
+          // Refresh local versions
+          case GlobalSetting.localVersion: {
+            this.refreshLocalVersions(getLocalVersions());
+            break;
+          }
+
+          case GlobalSetting.acceleratorsToBlock:
+          case GlobalSetting.channelsToShow:
+          case GlobalSetting.electronMirror:
+          case GlobalSetting.environmentVariables:
+          case GlobalSetting.executionFlags:
+          case GlobalSetting.fontFamily:
+          case GlobalSetting.fontSize:
+          case GlobalSetting.gitHubAvatarUrl:
+          case GlobalSetting.gitHubLogin:
+          case GlobalSetting.gitHubName:
+          case GlobalSetting.gitHubToken:
+          case GlobalSetting.hasShownTour:
```

Comment:
This is already handled elsewhere, so:

```suggestion
```

Reply:
Oddly making this change caused TypeScript to complain that about `this[key] = parsedValue;` assigning `unknown` to `never`. 🤔 There's some weird TS behavior around this bit (this PR actually pops out of me trying to upgrade TS), but I've added 2e19f042a85c786f3cda04d38d944f0120ccba2a to silence the TS error.